### DISCUSSION TITLE
MySQL to StarRocks data sync tool (link)

### DISCUSSION THREAD

**Discussion by alberttwong** [StarRocks/starrocks#26353] [How-tos, Tutorials, Best Practices and Architecture Articles]
# Preferred

This tutorial is to show how to quickly build a Streaming ELT job from MySQL to StarRocks using Flink CDC 3.0，including the feature of sync all table of one database, schema change evolution and sync sharding tables into one table.

https://ververica.github.io/flink-cdc-connectors/master/content/quickstart/mysql-starrocks-pipeline-tutorial.html

# Alternatives

This tool will allow you to sync MySQL to StarRocks or MongoDB to StarRocks - https://github.com/liuxinwang/go-mysql-starrocks 

There is also the StarRocks Migration Tool (SMT) that uses Apache Flink - https://docs.starrocks.io/docs/3.2/loading/Flink_cdc_load/#synchronize-database--table-schema

**Comment by kimmking** [comment]
we also can use Apache ShardingSphere to sync data from MySQL to SR, such as all table of one database, schema change evolution and sync sharding tables into one table.
